services/port_manager/src/main/java/com/futurewei/alcor/portmanager/repo/NeighborRepository.java

```diff
@@ -53,16 +55,21 @@ public void createNeighbors(Map<String, List<NeighborInfo>> neighbors) throws Ex
                         .stream()
                         .collect(Collectors.toMap(NeighborInfo::getPortIp, Function.identity()));
 
-                ICache<String, NeighborInfo> neighborCache = this.cacheFactory.getCache(
-                        NeighborInfo.class, getNeighborCacheName(entry.getKey()));
+                CacheConfiguration cfg = new CacheConfiguration();
+                cfg.setName(getNeighborCacheName(entry.getKey()));
+                cfg.setAtomicityMode(CacheAtomicityMode.TRANSACTIONAL);
```

Comment:
As this cache configuration is used in multiple places, could we abstract it in a util method in the common lib?

Reply:
I think a new API in our ignite module is better. New API will the same as what is now, getCache(name) will uncondionally create the cache in TXN mode and a new API getCacheNonTransactional(name) will behave like the current getCache, i.e, create ATMOIC cache. This will ensure that we are making minimal changes.  getCacheNonTransactional() will also make it clear that the cache is not required to be under TXN ever. It will also help us from inadvertanly mixing the two.

Reply:
Good. Thanks for your suggestions Prasad.

Reply:
I think @pkommoju's suggestion is good. Let us plan to change in next PR or so @DavidLiu506.